src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -763,10 +763,12 @@ private static bool IsExecutable(string fullPath)
                 return false;
             }
 
-            Interop.Sys.Permissions permissions = ((Interop.Sys.Permissions)fileinfo.Mode) & Interop.Sys.Permissions.S_IXUGO;
+            const UnixFileMode allExecute = UnixFileMode.UserExecute | UnixFileMode.GroupExecute | UnixFileMode.OtherExecute;
+
+            UnixFileMode permissions = (UnixFileMode)(fileinfo.Mode & (int)allExecute);
```

Comment:
```suggestion
            UnixFileMode permissions = ((UnixFileMode)fileinfo.Mode) & allExecute;
```

Gets rid of one cast.